projects/telemetry/src/gps.c

```diff
@@ -0,0 +1,78 @@
+#include "gps.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include "delay.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "uart.h"
+#include "xbee.h"
+
+// A large struct to store data and settings. Since the GPS should only be initialized once
+static GpsSettings *s_settings = NULL;
+
+// This method will be called every time the GPS sends data.
+static void prv_gps_callback(const uint8_t *rx_arr, size_t len, void *context) {
+  xbee_transmit(rx_arr, len);
```

Comment:
Why the direct passthrough here? I think it makes more sense to parse the NMEA sentences either in the GPS driver, or some other NMEA parsing module.

I don't think it makes sense to have the Telemetry software parse the NMEA sentences, since the way it's currently architected has it parsing encoded CAN messages. It would probably be easier to provision a new CAN message and use that instead.

Reply:
So the original plan was to parse messages on the telemetry board but I decided to just pass the data through since there wasn't any point in parsing it right away. I will refactor it and do the parsing on the board